update_manifest.py

```diff
@@ -1,56 +1,135 @@
+    try:
+        config.read("manifest.cfg")
+    except FileNotFoundError:
+        logging.critical(f"Manifest configuration file not found in path '{base_path}'")
+        return
 
-    for file in root.iter("File"):
-        name, part = file.get("name"), file.get("part")
-        path = parts.get(part) / name
-        try:
+    base_url = "https://raw.githubusercontent.com/PathOfBuildingCommunity/PathOfBuilding/{branch}/"
+    parts: list[dict[str, str]] = []
+    for part in config.sections():
+        url = base_url + config[part]["path"]
+        url_with_trailing_slash = url if url.endswith("/") else url + "/"
+        attributes = {"part": part, "url": url_with_trailing_slash}
+        if part == "runtime":
+            attributes.update(platform="win32")
+        parts.append(attributes)
+
+    rules = {
+        "include-files": (_complement, _exclude_file),
+        "include-directories": (_complement, _exclude_directory),
+        "exclude-files": (_identity, _exclude_file),
+        "exclude-directories": (_identity, _exclude_directory),
+    }
+    files: list[dict[str, str]] = []
+    for section in config.sections():
+        exclusion_checks = [
+            modifier(functools.partial(check, set(config[section][option].split(","))))
+            for option, (modifier, check) in rules.items()
+            if config.has_option(section, option)
+        ]
+        source = pathlib.Path(config[section]["path"])
+        for path in source.glob("**/*.*"):
+            if any(is_excluded(path) for is_excluded in exclusion_checks):
+                continue
             data = path.read_bytes()
-        except FileNotFoundError:
-            logger.error(f"File not found, skipping {path}")
-            continue
-        sha1_hash = hashlib.sha1(data).hexdigest()
-        file.set("sha1", sha1_hash)
-        logger.info(f"Path: {path} hash: {sha1_hash}")
-    if version is not None:
-        root.find("Version").set("number", version)
-        logger.info(f"Updated to version {version}")
+            sha1 = hashlib.sha1(data).hexdigest()
+            name = path.relative_to(config[section]["path"]).as_posix()
+            attributes = {"part": section, "sha1": sha1, "name": name}
+            if path.suffix in [".dll", ".exe"]:
+                attributes.update(runtime="win32")
+            files.append(attributes)
 
+    files.sort(key=lambda attr: (attr["part"], _alphanumeric(attr["name"])))
+
+    root = Et.Element("PoBVersion")
+    Et.SubElement(root, "Version", number=new_version)
+    for attributes in parts:
+        Et.SubElement(root, "Source", attributes)
+    for attributes in files:
+        Et.SubElement(root, "File", attributes)
     file_name = "manifest.xml" if replace else "manifest-updated.xml"
-    manifest.write(base_path / file_name, encoding="UTF-8", xml_declaration=True)
+    tree = Et.ElementTree(root)
+    tree.write(base_path / file_name, encoding="UTF-8", xml_declaration=True)
```

Comment:
I had to insert `Et.indent(tree, '\t')` here to fix the all-on-one-line issue I mentioned.